src/dmd/root/ctfloat.d

```diff
@@ -216,17 +216,8 @@ extern (C++) struct CTFloat
     }
 
     // Constant real values 0, 1, -1 and 0.5.
-    __gshared real_t zero;
-    __gshared real_t one;
-    __gshared real_t minusone;
-    __gshared real_t half;
-
-    @trusted
-    static void initialize()
-    {
-        zero = real_t(0);
-        one = real_t(1);
-        minusone = real_t(-1);
-        half = real_t(0.5);
-    }
+    __gshared const real_t zero = 0;
```

Comment:
This doesn't work with LDC on windows:
```
.\dmd\root\longdouble.d(293): error : `asm` statements cannot be interpreted at compile time [D:\a\1\s\src\vcbuild\dmd.vcxproj]
  ..\dmd\root\longdouble.d(91):        called from here: `ld_set(&this, cast(double)i)`
  ..\dmd\root\ctfloat.d(219):        called from here: `longdouble_soft(13835058055282163713LU, cast(ushort)32767u).this(0)`
..\dmd\root\longdouble.d(293): error : `asm` statements cannot be interpreted at compile time [D:\a\1\s\src\vcbuild\dmd.vcxproj]
  ..\dmd\root\longdouble.d(91):        called from here: `ld_set(&this, cast(double)i)`
  ..\dmd\root\ctfloat.d(220):        called from here: `longdouble_soft(13835058055282163713LU, cast(ushort)32767u).this(1)`
..\dmd\root\longdouble.d(293): error : `asm` statements cannot be interpreted at compile time [D:\a\1\s\src\vcbuild\dmd.vcxproj]
  ..\dmd\root\longdouble.d(91):        called from here: `ld_set(&this, cast(double)i)`
  ..\dmd\root\ctfloat.d(221):        called from here: `longdouble_soft(13835058055282163713LU, cast(ushort)32767u).this(-1)`
..\dmd\root\longdouble.d(293): error : `asm` statements cannot be interpreted at compile time [D:\a\1\s\src\vcbuild\dmd.vcxproj]
  ..\dmd\root\longdouble.d(105):        called from here: `ld_set(&this, d)`
  ..\dmd\root\ctfloat.d(222):        called from here: `longdouble_soft(13835058055282163713LU, cast(ushort)32767u).this(0.5)`
```
No idea why the azur CI doesn't fail due to this error.

Reply:
> I think the old initializers should be restored.

Aa an alternative intializers like these could be used: https://github.com/dlang/dmd/blob/master/src/dmd/root/longdouble.d#L663

Reply:
In fact, they could simply be aliased to those values in longdouble.d.

Reply:
Update to use the `longdouble` constructor. @rainers please confirm the values.

Reply:
Values look good, but it now reports:
```
D:\a\1\s\src\dmd\expression.d(849): Error: cannot implicitly convert expression `ld_zero` of type `const(longdouble_soft)` to `real`
D:\a\1\s\src\dmd\expression.d(855): Error: cannot implicitly convert expression `ld_zero` of type `const(longdouble_soft)` to `real`
D:\a\1\s\src\dmd\expression.d(861): Error: none of the overloads of `this` are callable using argument types `(const(longdouble_soft))`, candidates are:
D:\a\1\s\src\dmd\complex.d(22):        `dmd.complex.complex_t.this()`
D:\a\1\s\src\dmd\complex.d(24):        `dmd.complex.complex_t.this(real re)`
D:\a\1\s\src\dmd\complex.d(29):        `dmd.complex.complex_t.this(real re, real im)`
```
Looks like `real_t` is mapped to `real` instead of `longdouble` somewhere.

Reply:
> but it now reports:

Ah, this is not happening when the compiler is built, but the unittest runner. Could be the wrong version in that case.

Reply:
> No idea why the azur CI doesn't fail due to this error.

@wilzbach Any idea what's going on there? See for example the CmdLine task of "Windows_VisualD visuald-ldc" of https://dev.azure.com/dlanguage/dmd/_build/results?buildId=1465&view=logs

Reply:
Looks like `real_t` is set to `real` instead of `longdouble_soft`.

Reply:
Why is the test passing now?

Reply:
Due to the corrected static if. The test is compiled with the dmd just built, not LDC.

Reply:
I don't understand why the last change made it work. `longdouble_soft` is only available if `CRuntime_Microsoft` is true, regardless of the size of `real`.

Reply:
> the azur CI doesn't fail due to this error.

Looks like it isn't run. @jacob-carlborg have you tried rebasing the PR to the lastest `upstream/master`?

```
git fetch upstream
git rebase upstream/master
```

Reply:
Done.